dlrover/python/master/node/dist_job_manager.py

```diff
@@ -213,15 +214,37 @@ def _has_running_workers(self):
     def get_worker_num(self):
         return self._job_resource.worker_num
 
-    def early_stop(self):
-        nodes = self._ps_manager.get_pending_timeout_oom_recovered_node()
-        msg = ""
-        if len(nodes) > 0:
+    def should_early_stop(self):
+        # ps pending judgement: any ps pod pending timeout
+        timeout_ps_nodes = (
+            self._ps_manager.get_pending_timeout_oom_recovered_node()
+        )
+        if len(timeout_ps_nodes) > 0:
             msg = (
-                "Stop the training early because "
-                "OOM recoverd pod pends too long."
+                "Stop the training early because oom recovered node pending "
```

Comment:
the nodes recovered from OOM are pending too long and have timed out.

Reply:
done